### DISCUSSION TITLE
"Failed to execute 'importNode' on 'Document': parameter 1 is not of type 'Node'."

### DISCUSSION THREAD

**Discussion by captDaylight** [solidjs/solid#1709] [Q&A]
I have a repo with two packages, one that bundles up some custom elements and one that exports pure SolidJS components for use across the repo. My team has some existing code that uses Shoelace that broke when I imported components from our custom library.

Here's an example code snippet, when I add the `<sl-spinner>` (previously worked until I added the component lib) I get the error `Failed to execute 'importNode' on 'Document': parameter 1 is not of type 'Node'.`.

``` tsx
import '@shoelace-style/shoelace/dist/components/spinner/spinner';
import { setBasePath } from '@shoelace-style/shoelace/dist/utilities/base-path.js';
setBasePath('https://cdn.jsdelivr.net/npm/@shoelace-style/shoelace@2.4.0/dist/');

import { customElement } from 'solid-element';
import { MyComponent } from '@my-company/components';

customElement('my-web-component', {}, () => {
  return (
    <div>
      <sl-spinner style="font-size: 3rem;"></sl-spinner>
      <MyComponent />
    </div>
  );
});
```

When I try the code below, it displays "YES". This leads me to believe that the spinner is registered as a custom element:

``` tsx
<Show when={customElements.get('sl-spinner') !== undefined}>YES</Show>
```

But when I try this I again get the same error in the title:

``` tsx
<Show when={customElements.get('sl-spinner') !== undefined}>
    <sl-spinner></sl-spinner>
</Show>
```

UPDATE:
I got the Shoelace component to work (in a less than ideal way):

``` tsx
function SlSpinnerWrapper() {
  let parentEl: HTMLDivElement;

  createEffect(() => {
    if (customElements.get('sl-spinner')) {
      const spinnerElement = document.createElement('sl-spinner');
      if (parentEl) {
        parentEl.appendChild(spinnerElement);
      }
    }
  });

  return <div ref={parentEl}></div>;
}
```

Then I can use it like this 
``` tsx
<SlSpinnerWrapper />
```

Obviously this is not the way I want to use it, but I'm hoping it illustrates the issue that's going on.

**Comment by Fxxxlei** [comment]
I am having the same issue.
In the compiled code, it seems the output of `template(...)` to `document.inportNode`.
`template(...)` returns a function, but `document.inportNode` expects a Node, hence the issue.
When the output of a component contains only native elements, the generated code runs `template(...).cloneNode(true)` instead, which returns the correct Node.
Running `template(...).cloneNode(true)` for custom element gives the correct Node as well, so this is probably a bug?
